cmd/kubeadm/app/phases/bootstraptoken/clusterinfo/clusterinfo.go

```diff
@@ -108,3 +113,59 @@ func CreateClusterInfoRBACRules(client clientset.Interface) error {
 		},
 	})
 }
+
+var (
+	// tokenRegexpString defines id.secret regular expression pattern
+	tokenRegexpString = "^([a-z0-9]{6})\\.([a-z0-9]{16})$"
+	// tokenRegexp is a compiled regular expression of TokenRegexpString
+	tokenRegexp = regexp.MustCompile(tokenRegexpString)
+)
+
+// parseToken tries and parse a valid token from a string.
+// A token ID and token secret are returned in case of success, an error otherwise.
+func parseToken(s string) (string, string, error) {
```

Comment:
tests will be needed for those functions.

Reply:
I will be surprised if such function is not already implemented somewhere else....